datadog/resource_datadog_metric_metadata.go

```diff
@@ -95,8 +95,10 @@ func resourceDatadogMetricMetadataCreate(ctx context.Context, d *schema.Resource
 }
 
 func updateMetricMetadataState(d *schema.ResourceData, metadata *datadogV1.MetricMetadata) diag.Diagnostics {
-	if err := d.Set("type", metadata.GetType()); err != nil {
-		return diag.FromErr(err)
+	if _, ok := d.GetOk("type"); !ok || metadata.GetType() != "distribution" {
+		if err := d.Set("type", metadata.GetType()); err != nil {
+			return diag.FromErr(err)
+		}
```

Comment:
If I understand correctly, this change allows the type to be changed if the type is valid and if the type is not distribution.

That seems fine to me since `distribution` is feature flagged and  should not be editable

Reply:
That is correct